L1Trigger/TrackerTFP/plugins/ProducerKFin.cc

```diff
@@ -0,0 +1,207 @@
+        vector<deque<FrameTrack>> dequesTracks(dataFormats_->numChannel(Process::kf));
+        int i(0);
+        for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks) {
+          if ((int)ttTrack.phiSector() / setup_->numSectorsPhi() != region) {
+            i++;
+            continue;
+          }
+          const int sectorPhi = ttTrack.phiSector() % setup_->numSectorsPhi();
+          deque<FrameTrack>& tracks = dequesTracks[sectorPhi];
+          const int binEta = ttTrack.etaSector();
+          const int binZT = dfZT.toUnsigned(dfZT.integer(ttTrack.z0()));
+          const int binCot = dfCot.toUnsigned(dfCot.integer(ttTrack.tanL()));
+          StubSF* stubSF = nullptr;
+          TTBV hitPattern(0, setup_->numLayers());
+          vector<int> layerCounts(setup_->numLayers(), 0);
+          for (const TTStubRef& ttStubRef : ttTrack.getStubRefs()) {
+            const int layerId = setup_->layerId(ttStubRef);
+            const int layerIdKF = layerEncoding_->layerIdKF(binEta, binZT, binCot, layerId);
+            hitPattern.set(layerIdKF);
+            if (layerCounts[layerIdKF] == setup_->kfMaxStubsPerLayer())
+              continue;
+            layerCounts[layerIdKF]++;
+            deque<TTDTC::Frame>& stubs = dequesStubs[sectorPhi * setup_->numLayers() + layerIdKF];
+            auto identical = [ttStubRef, ttTrack](const StubSF& stub){
+              return (int)ttTrack.trackSeedType() == stub.trackId() && ttStubRef == stub.ttStubRef();
+            };
+            stubSF = &*find_if(stubsSF.begin(), stubsSF.end(), identical);
+            const StubKFin stubKFin(*stubSF, layerIdKF, ttTrack.hitPattern());
+            stubs.emplace_back(stubKFin.frame());
+          }
+          const TTBV& layerMap = setup_->layerMap(hitPattern, layerCounts);
+          const TTBV& maybePattern = layerEncoding_->maybePattern(binEta, binZT, binCot);
+          TrackKFin track(*stubSF, TTTrackRef(handleTTTracks, i++), hitPattern, layerMap, maybePattern);
+          tracks.emplace_back(track.frame());
+        }
+        // transform deques to vectors
+        for (int channel = 0; channel < dataFormats_->numChannel(Process::kf); channel++) {
+          const int index = region * dataFormats_->numChannel(Process::kf) + channel;
+          deque<FrameTrack>& tracks = dequesTracks[channel];
+          const auto limitTracks = next(tracks.begin(), min((int)tracks.size(), enableTruncation_ ? setup_->numFrames() : 0));
+          streamAcceptedTracks[index] = StreamTrack(tracks.begin(), limitTracks);
+          streamLostTracks[index] = StreamTrack(limitTracks, tracks.end());
+          for (int l = 0; l < setup_->numLayers(); l++) {
+            deque<TTDTC::Frame>& stubs = dequesStubs[channel * setup_->numLayers() + l];
+            const auto limitStubs = next(stubs.begin(), min((int)stubs.size(), enableTruncation_ ? setup_->numFrames() : 0));
```

Comment:
if enableTruncation_ is false, then L191 seems to set limitStubs to stubs.begin() instead of stubs.end().

Reply:
no.